source/game.c

```diff
@@ -2203,10 +2203,265 @@ static bool check_and_score_joker_for_event(ListItr* starting_joker_itr, CardObj
+        _joker_scored_itr = list_itr_create(&_owned_jokers_list);
+        scored_card_index = hand_top;
+    }
+
+    if (card_object_is_selected(played[i]))
+    {
+        (*played_y) -= int2fx(10);
+    }
+    return false;
+}
+
+// returns true if the scoring loop has returned early
+static bool play_scoring_held_update(int i, FIXED* played_y)
+{
+    if (i == 0 && (timer % FRAMES(30) == 0) && timer > FRAMES(40))
+    {
+        tte_erase_rect_wrapper(HELD_CARDS_SCORES_RECT);
+
+        // Go through all held cards and see if they activate Jokers
+        for ( ; scored_card_index >= 0; scored_card_index--)
+        {
+            if (check_and_score_joker_for_event(&_joker_scored_itr, hand[scored_card_index], JOKER_EVENT_ON_CARD_HELD))
+            {
+                card_object_shake(hand[scored_card_index], SFX_CARD_SELECT);
+                return true;
+            }
+            _joker_scored_itr = list_itr_create(&_owned_jokers_list);
+        }
+        scored_card_index = 0;
+
+        play_state = PLAY_SCORING_JOKERS;
+    }
+
+    if (card_object_is_selected(played[i]))
+    {
+        (*played_y) -= int2fx(10);
+    }
+    return false;
+}
+
+// returns true if the scoring loop has returned early
+static bool play_scoring_jokers_update(int i, FIXED* played_y)
+{
+    if (i == 0 && (timer % FRAMES(30) == 0) && timer > FRAMES(40))
+    {
+
+        tte_erase_rect_wrapper(PLAYED_CARDS_SCORES_RECT);
+
+        if (check_and_score_joker_for_event(&_joker_scored_itr, NULL, JOKER_EVENT_INDEPENDENT))
+        {
+            return true;
+        }
+
+        // Trigger hand end effect for all jokers once they are done scoring
+        if (check_and_score_joker_for_event(&_joker_round_end_itr, NULL, JOKER_EVENT_ON_HAND_SCORED_END))
+        {
+            return true;
+        }
+
+        play_state = PLAY_ENDING;
+        timer = TM_ZERO;
+        scored_card_index = played_top + 1; // Reset the scored card index to the top of the played stack
+        return false;
+    }
+
+    if (card_object_is_selected(played[i]))
+    {
+        (*played_y) -= int2fx(10);
+    }
+    return false;
+}
+
+static void play_ending_update(int i, FIXED* played_y)
+{
+    if (i == 0 && (timer % FRAMES(10) == 0 || !card_object_is_selected(played[played_top - scored_card_index])) && timer > FRAMES(40))
+    {
+        scored_card_index--;
+
+        if (scored_card_index == 0)
+        {
+            play_state = PLAY_ENDED;
+            timer = TM_ZERO;
+        }
+    }
+
+    if (card_object_is_selected(played[i]) && played_top - i <= scored_card_index - 1)
+    {
+        (*played_y) -= int2fx(10);
+    }
+}
+
+static void play_ended_update(bool* discarded_card, bool* sound_played, int i, FIXED* played_x, FIXED* played_y)
 {
-    // TODO: Break this function up into smaller ones.
+    if (!*discarded_card && played[i] != NULL && timer > FRAMES(40))
+    {
+        (*played_x) = int2fx(240);
+        (*played_y) = int2fx(70);
```

Comment:
```suggestion
        (*played_x) = int2fx(CARD_DISCARD_PNT.x);
        (*played_y) = int2fx(CARD_DISCARD_PNT.y);
```
Since we're at it, let's remove this magic number, there's already an existing constant for it.